pkg/recipes/terraform/config/config.go

```diff
@@ -150,13 +167,24 @@ func getProviderConfigs(ctx context.Context, requiredProviders []string, support
 			return nil, err
 		}
 		if len(config) > 0 {
-			providerConfigs[provider] = config
+			providerConfigs[provider] = []any{config}
 		}
 	}
 
 	return providerConfigs, nil
 }
 
+// getRecipeProviderConfigs returns the Terraform provider configurations for Terraform providers.
+func getRecipeProviderConfigs(ctx context.Context, envConfig *recipes.Configuration) map[string][]any {
+	// If the provider is not configured, return an empty config
+	if envConfig.RecipeConfig.Terraform.Providers.ProviderConfigs == nil {
+		// Handle the case where one or more objects in the chain are nil...
+		return nil
```

Comment:
This should return an empty map instead. In current this if condition isn't really doing anything, this return statement would result into same value as L185.